From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Clark <tjclark@wpi.edu>
Date: Fri, 15 Aug 2014 16:21:45 -0400
Subject: [PATCH 0300/6262] Make Java tests run with TestNG

TestNG generates xml reports that Jenkins can use, JUnit does not without
Ant or Maven.

To run the tests, deploy the Jar and run:
    /usr/local/frc/JRE/bin/java -jar wpilibJavaIntegrationTests-0.1.0-SNAPSHOT.jar -junit -testclass edu.wpi.first.wpilibj.test.TestSuite

Change-Id: I9e8a7dcab775f9c5ad1905afa798cd9191e25438
---
 wpilibj/wpilibJavaIntegrationTests/pom.xml | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/wpilibj/wpilibJavaIntegrationTests/pom.xml b/wpilibj/wpilibJavaIntegrationTests/pom.xml
index 2718621ed44b57f6f8b7b9cf997bce897685494e..f2bb1034d42aa883bbdb2890e74a0c3f2a3d809d 100644
--- a/wpilibj/wpilibJavaIntegrationTests/pom.xml
+++ b/wpilibj/wpilibJavaIntegrationTests/pom.xml
@@ -34,6 +34,11 @@
 			<artifactId>junit-toolbox</artifactId>
 			<version>2.0</version>
 		</dependency>
+		<dependency>
+			<groupId>org.testng</groupId>
+			<artifactId>testng</artifactId>
+			<version>6.8.8</version>
+		</dependency>
 	</dependencies>
 
 	<profiles>
@@ -120,6 +125,12 @@
 									<version>2.0</version>
 									<type>jar</type>
 								</artifactItem>
+								<artifactItem>
+									<groupId>org.testng</groupId>
+									<artifactId>testng</artifactId>
+									<version>6.8.8</version>
+									<type>jar</type>
+								</artifactItem>
 							</artifactItems>
 							<outputDirectory>${project.build.directory}/classes</outputDirectory>
 						</configuration>
@@ -148,7 +159,7 @@
 					<archive>
 						<manifest>
 							<addClasspath>true</addClasspath>
-							<mainClass>edu.wpi.first.wpilibj.test.TestSuite</mainClass>
+							<mainClass>org.testng.TestNG</mainClass>
 						</manifest>
 					</archive>
 				</configuration>
